homeassistant/components/xiaomi_miio/sensor.py

```diff
@@ -658,6 +661,26 @@ def __init__(self, name, device, entry, unique_id, coordinator, description):
         self._attr_native_value = self._determine_native_value()
         self._attr_extra_state_attributes = self._extract_attributes(coordinator.data)
 
+    def _parse_time_delta(self, value: timedelta) -> float:
+        """Overridden to handle different time units for timedelta.
+
+        This ensures that the device reported native values are converted to the ones
+        as defined in the entity descriptions.
+
+        This needs to be done here as the parent class does not have access
+        to the entity description.
+        """
+        conversion_map = {
+            TIME_SECONDS: timedelta(seconds=1),
+            TIME_MINUTES: timedelta(minutes=1),
+            TIME_HOURS: timedelta(hours=1),
+            TIME_DAYS: timedelta(days=1),
+        }
+
+        wanted_unit = self.entity_description.native_unit_of_measurement
+
+        return round((value / conversion_map[wanted_unit]), 2)
```

Comment:
Besides this being a major breaking change, we should not calculate this inside Home Assistant.

Home Assistant should represent data as is.

Reply:
So I suppose we should close this?

I agree that this is not the task of the integration while I understand that it's frustrating to users to create separate template sensors or adjust automations (or the UI display) to do the conversions.

Reply:
> I agree that this is not the task of the integration while I understand that it's frustrating to users to create separate template sensors or adjust automations (or the UI display) to do the conversions.

Yeah, it is a hard balance, as in: We all have preferences, right?

Reply:
We should remember that this whole discussion started when the upstream package suddenly provided seconds instead of hours for these sensors.

It used to be hours 🤷🏻

Reply:
Time is the only thing we allow converting to fit our design. If we want to convert it to lower resolution I think that would be ok.

Reply:
> Yeah, it is a hard balance, as in: We all have preferences, right?

Right. I think the problem at hand is the fact that while we have native measurement units, we are lacking a way to inform the UI the unit they should be using for displaying the data (or to configure the display unit, for that matter). So maybe all entities should be using the most precise format the device gives, while allowing the integrations to add "unit hints" to signal their preferred display format for UI elements? This discussion doesn't belong to this PR though, so I'll leave it to that and I'll close this PR now.

> I'd like to recall that this whole discussion started when the upstream package suddenly provided seconds instead of hours for these sensors.

That's not true, the upstream package has always delivered seconds. The earlier homeassistant implementation converted the values for display reasons, which was necessary as there was no facilities in homeassistant to report on the used unit (as we have nowadays).